htdocs/api/v0.0.3-dev/Candidates.php

```diff
@@ -115,47 +115,73 @@ public function handlePOST()
             $this->error("There is no Candidate object in the POST data");
             $this->safeExit(0);
         }
-
-        // This version of the API does not handle candidate creation
-        // when users are at multiple sites
-        $user      = \User::singleton();
-        $centerIDs = $user->getCenterIDs();
-        $num_sites = count($centerIDs);
-
+        // When users are at multiple sites, the API requires
+        // siteName as an input to candidate creation
+        $user         = \User::singleton();
+        $centerIDs    = $user->getCenterIDs();
+        $allUserSites = array();
+        $num_sites    = count($centerIDs);
         if ($num_sites == 0) {
             $this->header("HTTP/1.1 401 Unauthorized");
             $this->error("You are not affiliated with any site");
             $this->safeExit(0);
-        }
+        } else if ($num_sites > 1) {
+            foreach ($centerIDs as $key => $centerID) {
+                $center = $this->DB->pselectRow(
+                    "SELECT CenterID as ID, Name FROM psc WHERE CenterID =:cid",
+                    array('cid' => $centerID)
+                );
+                $allUserSites[$centerID] = $center['Name'];
+            }
 
-        if ($num_sites > 1) {
-            $this->header("HTTP/1.1 501 Not Implemented");
-            $this->error(
-                "This API version does not support candidate creation " .
-                "by users with multiple site affiliations. This will be ".
-                "implemented in a future release."
-            );
-            $this->safeExit(0);
-        }
+            $siteName = $data['Candidate']['Site'];
+            $this->verifyField($data, 'Site', $allUserSites);
+            $this->verifyField($data, 'Gender', ['Male', 'Female']);
+            $this->verifyField($data, 'EDC', 'YYYY-MM-DD');
+            $this->verifyField($data, 'DoB', 'YYYY-MM-DD');
+            //Get the CenterID from the provided SiteName
+            $centerID = array_search($siteName, $allUserSites);
```

Comment:
here maybe you can do an explicit check for `$centerID === FALSE` to throw an error that the user is not affiliated with the site they provided / the site name they provided is invalid
also maybe doing a check that the site name is a valid one first could be useful to discern the two aforementioned errors? e.g., problem with permissions / site affiliation vs. just making a typo

Reply:
I did split the 2 errors. Now I get a list of all the sitenames and check that the provided one is in the array (otherwise a 400 error), and if it is, then I check if it is in the list of the USER's sitenames, and if not, I give the 403 error